### PR TITLE
Moved to buf; Added buf lint; Fixed ping service to match standards; …

### PR DESCRIPTION
**Pull request by bwplotka**
…Moved auth to interceptors.

Chained with https://github.com/grpc-ecosystem/go-grpc-middleware/pull/321

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
PTAL @johanbrandhorst @yashrsharma44

File: Makefile [hunk 1]
```diff
@@ -98,11 +92,21 @@ lint_module: $(FAILLINT) $(GOLANGCI_LINT) $(MISSPELL)
...
-$(PROTOC):
+
+# For protoc naming matters.
+PROTOC_GEN_GO_CURRENT := $(TMP_GOPATH)/protoc-gen-go
+PROTOC_GEN_GO_GRPC_CURRENT := $(TMP_GOPATH)/protoc-gen-go-grpc
+PROTO_TEST_DIR := testing/testpb/v1
+
+.PHONY: proto
+proto: ## Generate testing protobufs
+proto: $(BUF) $(PROTOC_GEN_GO) $(PROTOC_GEN_GO_GRPC) $(PROTO_TEST_DIR)/test.proto
```
Comment (johanbrandhorst) [inline Makefile:103]: Should we have two targets name `proto`? I'm not familiar with this Makefile pattern, if it is one.

File: Makefile [hunk 2]
```diff
@@ -98,11 +92,21 @@ lint_module: $(FAILLINT) $(GOLANGCI_LINT) $(MISSPELL)
...
-	@mv -- "$(TMP_GOPATH)/bin/protoc" "$(GOBIN)/protoc-$(PROTOC_VERSION)"
-	@echo ">> produced $(GOBIN)/protoc-$(PROTOC_VERSION)"
+	@cp $(PROTOC_GEN_GO) $(PROTOC_GEN_GO_CURRENT)
+	@cp $(PROTOC_GEN_GO_GRPC) $(PROTOC_GEN_GO_GRPC_CURRENT)
+	@echo ">> generating $(PROTO_TEST_DIR)"
+	@PATH=$(GOBIN):$(TMP_GOPATH) $(BUF) protoc \
+		-I $(PROTO_TEST_DIR) \
+		--go_out=$(PROTO_TEST_DIR)/../ \
+		--go-grpc_out=$(PROTO_TEST_DIR)/../ \
+	    $(PROTO_TEST_DIR)/*.proto
```
Comment (johanbrandhorst) [inline Makefile:112]: This is awesome, next step would be to just use `buf generate`: https://docs.buf.build/tour-6/. If you're feeling keen it should be possible to do here already with `buf.gen.yaml`:

```yaml
version: v1beta1
plugins:
  - name: go
    out: testing/testpb/v1
  - name: go-grpc
    out: testing/testpb/v1
```

You may want `opt: paths=source_relative` if the files aren't going in the right place.

**Review by johanbrandhorst — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (bwplotka) [reply Makefile:103]: I can see this being confusing, but it's totally ok. It's to ensure `make help` works (and shows this command)

File: Makefile (hunk 2 above)
Comment (bwplotka) [reply Makefile:112]: uuu nice, let's try it (or in sep PR)

File: testing/testpb/v1/test.proto [hunk 3]
```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package testing.testpb.v1;
```
Comment (yashrsharma44) [inline testing/testpb/v1/test.proto:3]: Looks solid 💪

**Review by yashrsharma44 — APPROVED** [review]
Looks good! 😎

File: .bingo/Variables.mk [hunk 4]
```diff
@@ -23,6 +23,12 @@ $(BINGO): $(BINGO_DIR)/bingo.mod
 	@echo "(re)installing $(GOBIN)/bingo-v0.3.0"
 	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=bingo.mod -o=$(GOBIN)/bingo-v0.3.0 "github.com/bwplotka/bingo"
 
+BUF := $(GOBIN)/buf-v0.35.1
```
Comment (yashrsharma44) [inline .bingo/Variables.mk:26]: Why are we adding buf dependency? Wanted to know for info purpose 😛

Comment (bwplotka) [reply .bingo/Variables.mk:26]: Because we use now buf as a tool! (:

**Comment by bwplotka** [comment]
Look like this is ok to be merged (: 

Let's make sure we merge after https://github.com/grpc-ecosystem/go-grpc-middleware/pull/321 is approved

**Comment by bwplotka** [comment]
Rebased, rdy to merge (:

**Pull request merged**